src/main.rs

```diff
@@ -228,12 +228,12 @@ fn handle_trap(ctx: &mut VirtContext, mctx: &mut MiralisContext, policy: &mut Po
         (ExecutionMode::Firmware, ExecutionMode::Payload) => {
             log::debug!("Execution mode: Firmware -> Payload");
             unsafe { ctx.switch_from_firmware_to_payload(mctx) };
-            policy.switch_from_firmware_to_payload(ctx);
+            policy.switch_from_firmware_to_payload(ctx, mctx);
```

Comment:
Ha, I see the PMPs are in the `mctx`. Let's continue with that, but it might be good to keep in mind how to cleanly separate the Miralis internal state from what the Policy need.